src/styles/createPalette.js

```diff
@@ -76,19 +76,37 @@ export const dark = {
   },
 };
 
-function getContrastText(hue) {
-  if (getContrastRatio(hue, common.black) < 7) {
-    return dark.text.primary;
-  }
-  return light.text.primary;
-}
-
 export default function createPalette(palette: Object) {
-  const { primary = indigo, secondary = pink, error = red, type = 'light', ...other } = palette;
+  const {
+    primary = indigo,
+    secondary = pink,
+    error = red,
+    type = 'light',
+    contrastThreshold = 0.29,
+    ...other
+  } = palette;
   const shades = { dark, light };
 
   warning(Boolean(shades[type]), `Material-UI: the palette type \`${type}\` is not supported.`);
 
+  function getContrastText(background) {
+    const contrastText =
+      getLuminance(background) <= contrastThreshold ? dark.text.primary : light.text.primary;
```

Comment:
I'm wondering if we should revert to the `getContrastRatio` approach, for a cleaner `defaultThreshold`? (The original reason behing this change was based on a false premise.)

Reply:
I agree, the logic should stay in the contrastRation domain, and not leak to the luminanc one
- https://github.com/twbs/bootstrap/blob/1d6e3710dd447de1a200f29e8fa521f8a0908f70/scss/_functions.scss#L59
- https://github.com/material-components/material-components-web/blob/ac46b8863c4dab9fc22c4c662dc6bd1b65dd652f/packages/mdc-theme/_functions.scss#L51